lib/libnvpair/libnvpair_json.c

```diff
@@ -27,6 +27,8 @@
 			return (-1);			\
 	} while (0)
 
+#define IS_UNSIGNED(tp) ((tp)0 - 1 > 0)
```

Comment:
You can cast `-1` directly.  Don't need to subtract to 0.

And you can use a slightly better version which accepts a variable:

```
#define is_unsigned(x)  (((__typeof__(x)) -1) > 0)
```

So you can use `c` directly:

```
if (((IS_UNSIGNED(c) || c >= 0x00) && c <= 0x1f) || (c > 0x7f && c <= 0xffff))
```

Reply:
I can also not to that because it doesn't work. And neither does any other suppression method, as noted in the original thread https://github.com/openzfs/zfs/pull/13110#discussion_r808488078